confluent_kafka_helpers/tracing/backends/opentracer.py

```diff
@@ -0,0 +1,92 @@
+import contextlib
+import io
+import sys
+import traceback
+
+import opentracing
+
+from confluent_kafka_helpers.tracing import tags
+
+DEFAULT_TAGS = {
+    tags.COMPONENT: 'confluent_kafka_helpers',
+    tags.PEER_SERVICE: 'kafka',
+}
+
+
+class OpenTracerBackend:
+    def __init__(self, tracer=opentracing.global_tracer):
+        self._tracer = tracer()
+
+    def __getattr__(self, name):
+        return getattr(self._tracer, name)
+
+    @contextlib.contextmanager
+    def start_active_span(self, *args, **kwargs):
+        # we can't use `start_active_span` or `start_span` built in context
+        # managers since they default catches all exceptions and logs them
+        # incorrectly in `opentracing.span.Span.__exit__` when using the
+        # Datadog tracer.
+        #
+        # https://docs.datadoghq.com/tracing/visualization/trace/?tab=spantags
+        try:
+            scope = self._tracer.start_active_span(
+                *args, tags=DEFAULT_TAGS, **kwargs
+            )
+            yield scope.span
+        finally:
+            self.log_exception(span=scope.span)
+            scope.span.finish()
+
+    @contextlib.contextmanager
+    def start_span(self, *args, **kwargs):
+        try:
+            span = self._tracer.start_span(*args, tags=DEFAULT_TAGS, **kwargs)
+            yield span
+        finally:
+            self.log_exception(span=span)
+            span.finish()
+
+    def extract_headers_and_start_span(self, operation_name, headers):
+        try:
+            parent_context = self._tracer.extract(
+                carrier=headers, format=opentracing.Format.TEXT_MAP
+            )
+        except (
+            opentracing.InvalidCarrierException,
+            opentracing.SpanContextCorruptedException,
+        ):
+            span = self.start_active_span(operation_name=operation_name)
```

Comment:
Just a clarification, we don't send in anything related to the headers here because we only want to extract header data from an existing span if it exists. The headers are irrelevant if they don't contain any tracing information?

Reply:
Yep exactly! If the tracing headers doesn't exist there's no previous trace to continue on so we can just start a fresh trace (first span creates the trace).

Reply:
👍